crates/editor/src/clangd_ext.rs

```diff
@@ -0,0 +1,119 @@
+use std::path::PathBuf;
+
+use anyhow::Context as _;
+use gpui::{View, ViewContext, WindowContext};
+use language::Language;
+use url::Url;
+
+use crate::{element::register_action, Editor, SwitchSourceHeader};
+
+pub fn apply_related_actions(editor: &View<Editor>, cx: &mut WindowContext) {
+    let is_c_related = editor.update(cx, |editor, cx| {
+        editor
+            .buffer()
+            .read(cx)
+            .all_buffers()
+            .iter()
+            .any(|b| match b.read(cx).language() {
+                Some(l) => is_c_language(l),
+                None => false,
+            })
+    });
+
+    if is_c_related {
+        register_action(editor, cx, switch_source_header);
+    }
+}
+
+pub fn switch_source_header(
+    editor: &mut Editor,
+    _: &SwitchSourceHeader,
+    cx: &mut ViewContext<'_, Editor>,
+) {
+    log::info!("switch_source_header action");
+
+    let Some(project) = &editor.project else {
+        return;
+    };
+    let Some(workspace) = editor.workspace() else {
+        return;
+    };
+
+    let multibuffer = editor.buffer().read(cx);
+
+    let Some((server_to_query, buffer)) = editor
+        .selections
+        .disjoint_anchors()
+        .into_iter()
+        .filter(|selection| selection.start == selection.end)
+        .filter_map(|selection| Some((selection.start.buffer_id?, selection.start)))
+        .filter_map(|(buffer_id, trigger_anchor)| {
+            let buffer = multibuffer.buffer(buffer_id)?;
+            let c_language = buffer.read(cx).language_at(trigger_anchor.text_anchor)?;
+            if !is_c_language(&c_language) {
+                return None;
+            }
+            Some(buffer)
+        })
+        .find_map(|buffer| {
+            project
+                .read(cx)
+                .language_servers_for_buffer(buffer.read(cx), cx)
+                .find_map(|(adapter, server)| {
+                    log::info!("adapter.name.0: {:?}", adapter.name.0);
```

Comment:
This too, feels like all new `log::` calls are debug-ones.
You can use `dbg!` (or `dbg!(&)`) for similar purposes and our Clippy profile will fail when run on CI.

Reply:
Thanks for pointing out the `dbg!` rules. Yes, I usually leave way too many debug prints in my code :sweat_smile: